java/build.gradle.kts

```diff
@@ -13,10 +13,12 @@ dependencies {
     api(projects.xefCore)
     api(projects.xefOpenai)
     api(projects.xefPdf)
+    api(projects.xefGpt4all)
```

Comment:
xef-java should not depend on the GPT4All integration. I need to see a table or schema on how you are all planning these integrations because currently the dependencies associations in this PR do not match the plans on making Java as lean as possible and not depend on another module that is not `xef-core` and Java specifics deps.

A module like `xef-java` should not depend on the rest of the modules. If another module like `xef-gpt4all` written in Kotlin is not properly encoded to be consumed from Java then it should be fixed there instead of bringing it as a dependency. The idea is that users of `xef-kotlin`, `xef-java` or `xef-scala` can import `xef-gpt4all` as dependency and use it without rewriting its apis. We don't want to port every API we develop in modules like `xef-gpt4All` or `xef-openai` to every language but instead, use them where possible. 

In the case of Java, Kotlin multiplatform has all the tools to export the types through @JvmName, @JvmStatic and @JvmFields as things that can be easily consumed from Java and other JVM langs like Scala. Only functions from `xef-core` should be ported to Java where needed if they use continuations or `suspend` for Kotlin. // cc @franciscodr @nomisRev for thoughts on how this should work.

Reply:
I agree, but not sure it will fit all use-cases. I.e. `Flow` is another one that might need wrapping and `@JvmName` or other Java annotations won't get use far enough.

We can of course expose Java specific implementation from `jvmMain`, but the biggest downside is that they will appear as extension functions and not class methods. Wrapping gives us a better and nicer APIs in those areas. I'm kind-of missing [C# partial classes](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods) for KMP implementations. Could make an interesting KEEP.

Scala has a similar issue where we might want to expose FS2 instead of `Flow`, or a different streaming data type. So we might end up wrapping some of the APIs, and exposing them from different modules to adapt the return types. However the underlying implementation would still be common.

Reply:
if I remove api(projects.xefGpt4all) from xef-java I can not import GPT4All and Gpt4AllModel and they are needed

Reply:
Also I could impl a promptStreaming in Chat.kt from xef-core but I would have to add the Publisher dependency, or add the implementation in CoreAIScope.kt but I would have to add the GPT4All dependency

@AiDsl
  suspend fun promptStreaming(
    question: String,
    context: VectorStore,
    promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS
  ): Publisher<String> =
    promptStreaming(Prompt(question), context, null, emptyList(), promptConfiguration).asPublisher()

Reply:
`projects.xefGpt4all` should only be added as a dependency in the examples.
xef-java should add the dependency for Reactive streams and implement `promptStreaming` in xef-java.

Reply:
Done